Build cannot find `openssl/sha.h`, while cmake does

I tried to build CCN-lite on macOS (I know there are issues, but still), but it fails during build due to a missing `openssl/sha.h` header. However, I have openssl installed via _homebrew_ and ran `cmake` successfully as follows:

```
cd ccn-lite/src
mkdir build
cd build
OPENSSL_ROOT_DIR=/usr/local/opt/openssl OPENSSL_INCLUDE_DIR=/usr/local/opt/openssl/include cmake ../
make
```

as usual there are some warnings about experimental ethernet support on macOS, but it finally fails due to openssl, with:

```
/ccn-lite/src/ccnl-core/include/ccnl-sockunion.h:51:9: warning: ethernet support in FreeBSD and MacOS is work in
      progress and not fully implemented! [-W#pragma-messages]
#pragma message "ethernet support in FreeBSD and MacOS is work in progress \
        ^
/ccn-lite/src/ccnl-core/src/ccnl-prefix.c:37:10: fatal error: 'openssl/sha.h' file not found
#include <openssl/sha.h>
         ^~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

This is discouraged by `brew` saying:

```
brew link openssl
Warning: Refusing to link: openssl
Linking keg-only openssl means you may end up linking against the insecure,
deprecated system OpenSSL while using the headers from Homebrew's openssl.
Instead, pass the full include/library paths to your compiler e.g.:
  -I/usr/local/opt/openssl/include -L/usr/local/opt/openssl/lib
```

and I also think the problem is somewhat different, because when compiling `ccnl-core/ccnl-prefix.c` the header file `openssl/sha.h` is found. To verify this I added a preprocessor warning directly below the include:

```C
#if !defined(CCNL_RIOT) && !defined(CCNL_ANDROID)
#warning huhu using openssl
#include <openssl/sha.h>
#endif // !defined(CCNL_RIOT) && !defined(CCNL_ANDROID)
```

and during build process I saw:

```cmake
[ 23%] Building C object ccnl-core/CMakeFiles/ccnl-core.dir/src/ccnl-prefix.c.o

/ccn-lite/src/ccnl-core/src/ccnl-prefix.c:37:2: warning: huhu using openssl [-W#warnings]
#warning huhu using openssl
 ^
2 warnings generated.
```

So I think the problem is with compiling `ccnl-unix`, and should be fixed there.

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -24,7 +24,7 @@ set(CCNL_BASIC_FLAGS
 add_definitions(${CCNL_BASIC_FLAGS})
 
 if (NOT DEFINED CCNL_RIOT)
-   set(CCNL_EXTRA_FLAGS 
+   set(CCNL_EXTRA_FLAGS
         -DUSE_CCNxDIGEST
         -DUSE_MGMT
         -DUSE_UNIXSOCKET
@@ -88,6 +88,13 @@ else()
 endif()
 
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS} -g")
+
+if (NOT DEFINED CCNL_LINUXKERNEL AND NOT DEFINED CCNL_RIOT)
+    find_package(OpenSSL REQUIRED)
+    include_directories(${OPENSSL_INCLUDE_DIR})
+    message("OpenSSL include dir: ${OPENSSL_INCLUDE_DIR}")
+    message("OpenSSL libraries: ${OPENSSL_LIBRARIES}")
+endif()
 
 #add_subdirectory(ccnl-addons)
 #if (DEFINED CCNL_RIOT)

--- a/src/ccnl-utils/CMakeLists.txt
+++ b/src/ccnl-utils/CMakeLists.txt
@@ -16,11 +16,7 @@ include_directories(
     ../ccnl-unix/include
     ../ccnl-nfn/include
     ../ccnl-pkt-compression/include
-    ${OPENSSL_INCLUDE_DIR}
-    
 )
-
-find_package(OpenSSL REQUIRED)
 
 add_executable(ccn-lite-peek ccn-lite-peek.c)
 #add_executable(ccn-lite-peekcomputation ccn-lite-peekcomputation.c) #todo work to do
@@ -56,7 +52,7 @@ if(OpenSSL_FOUND)
 
     target_link_libraries(ccn-lite-cryptoserver ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS} ${OPENSSL_LIBRARIES})
     target_link_libraries(ccn-lite-cryptoserver ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression ${OPENSSL_LIBRARIES})
-    
+
     #target_link_libraries(ccn-lite-deF ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS} ${OPENSSL_LIBRARIES})
     #target_link_libraries(ccn-lite-deF ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ${OPENSSL_LIBRARIES})
 
@@ -79,8 +75,8 @@ if(USE_FRAG)
     target_link_libraries(ccn-lite-mkF ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression)
 endif()
 
-target_link_libraries(ccn-lite-mkI ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS}) 
-target_link_libraries(ccn-lite-mkI ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression) 
+target_link_libraries(ccn-lite-mkI ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS})
+target_link_libraries(ccn-lite-mkI ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression)
 
 target_link_libraries(ccn-lite-pktdump ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS})
 target_link_libraries(ccn-lite-pktdump ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression)
@@ -90,7 +86,3 @@ target_link_libraries(ccn-lite-produce ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccn
 
 target_link_libraries(ccn-lite-simplenfn ${PROJECT_LINK_LIBS} ${EXT_LINK_LIBS})
 target_link_libraries(ccn-lite-simplenfn ccnl-core ccnl-pkt ccnl-fwd ccnl-unix ccnl-nfn ccnl-pkt-compression)
-
-
-
-

